### ISSUE TITLE
Help tmc_uart

### ISSUE THREAD

**Issue by BaltasarParreira** [Sergey1560/fb4s_howto#3]
Hi, I followed your UART guide witch is the best I found so far for the MKS Robin Nano v1.2 board. Excellent job here with all the provided information.
So I have a Robin Nano like that too and recently I upgraded it with 4 x TMC2209 v3.1 from fysetc, this ones here:
https://wiki.fysetc.com/Silent2209/#product-introduction

Using your "how to" I manage to put everything working but only using one pin (PA6) when I try to use two separated pins, one for TX and other for RX I get always the Marlin fail message "TMC Connection error" at the display on booting and can't use UART mode.
Also sending the command M122 I get this:
```
Driver registers:
		X	0x00:00:00:00	 Bad response!
		Y	0x00:00:00:00	 Bad response!
		Z	0x00:00:00:00	 Bad response!
		E	0x00:00:00:00	 Bad response!
Testing X connection... Error: All LOW
Testing Y connection... Error: All LOW
Testing Z connection... Error: All LOW
Testing E connection... Error: All LOW
```

I know that these drivers are a bit different from yours but I connected them like doing a cable with all four TX exits from the drivers together straight to PA6 and all the RX exits to PA1.
I'm not using the 1K resistor on TX line since is already internally on the driver as you can see from the v3.1 chematique on the driver link.

As I said not using the RX wire and setup all axis to use PA6 bought for TX and RX on Marlin configuration all runs well.

So my questions are:
- Is it ok to work like this since is working and until now I didn't see any problems or it is better really to have TX and RX separated using two pins? Can you see any problem on my wiring for theses TMC2209 drivers?
- Second is regarding the 800mA current since the previous stock drivers came with a VREF voltage of 0,81v witch gives lower current if is it safe to keep using the 800mA as described?

Thanks.

**Comment by Sergey1560** [comment]
Hello. 
By the scheme, the driver already has a 1k resistor. 
Have you tried swapping RX and TX? It works for me with 2 pins for 4 drivers, but during tests I tried with 1 pin and it also works.
If it works now, I see no reason to change something :)

On the second question - the current depends on the motors used. You do not need to calculate the voltage, you can specify the desired current directly. The higher the current - more engine heats up. If the current is insufficient, the motor skips steps. You need to find a balance.

**Comment by BaltasarParreira** [comment]
Hi, thanks for replying.
My printer is a Two Trees Bluer V2, I think the motors are this one (Usongshine US-17HS4401S):
![image](https://user-images.githubusercontent.com/1650425/115702512-f060dd80-a360-11eb-9e9d-937c3955866f.png)

When you say swap, you mean using PA6 for RX and PA1 for TX?

I did a couple of more tests and did a Y cable (exactly like your on the photo) from the RX driver pin (the one that doesn't pass trough the internal 1K resistor), so from this pin straight to the Robin Nano RX pin (PA1) and the other Y side trough a 1K resistor to the Robin Nano TX pin (PA6) and didn't work.
Second test was using only one pin for bought TX and RX but using my cable with the 1K resistor, so I connected my cable from the same driver RX pin (again the one without the internal 1K resistor) to the Robin Nano PA6 and didn't work.

So final conclusion, I only manage to have it working either using a direct cable from the TX or RX driver pins straight to the PA6 or PA1 or any other Robin Nano free pin without any resistor on this cable.
